app/Platform/Services/PlayerGameActivityService.php

```diff
@@ -61,7 +63,35 @@ public function initialize(User $user, Game $game, bool $withSubsets = false): v
             $gameIds[] = $game->id;
         }
 
-        $playerSessions = $user->playerSessions()->with('gameHash')->whereIn('game_id', $gameIds)->orderBy('created_at')->get();
+        // Get the most recent reset for this user and game(s).
+        // If the player has a recent reset for any of the games, we only
+        // want to include sessions from after the most recent reset.
+        // When `$withSubsets` is true, we need to check all gameIds.
+        $lastReset = null;
+        if ($withSubsets && count($gameIds) > 1) {
+            // If we're dealing with subsets, check for resets on any of the games.
+            $lastReset = PlayerProgressReset::where('user_id', $user->id)
+                ->where(function ($q) use ($gameIds) {
+                    $q->where(function ($subQuery) use ($gameIds) {
+                        $subQuery->where('type', PlayerProgressResetType::Game)
+                            ->whereIn('type_id', $gameIds);
+                    })
+                    ->orWhere('type', PlayerProgressResetType::Account);
+                })
+                ->orderByDesc('created_at')
+                ->first();
+        } else {
+            $lastReset = PlayerProgressReset::forUserAndGame($user, $game)->first();
+        }
+
+        $playerSessionsQuery = $user->playerSessions()
+            ->with('gameHash')
+            ->whereIn('game_id', $gameIds);
+        if ($lastReset) {
+            $playerSessionsQuery->where('created_at', '>=', $lastReset->created_at);
```

Comment:
We don't want to do this here. It removes all history before the reset from the game activity log for the user. There should still be empty sessions shown.

i.e. go to http://localhost:64000/user/Jamiras/game/472/activity
Reset progress for the game
Refresh the page - all sessions are missing.

`$lastReset` should only be used in `getAchievementSetMetrics` as a more restrictive start date than `$achievementsPublishedAt`.

Reply:
Updated in latest:
<img width="1261" height="175" alt="Screenshot 2025-07-16 at 6 01 10 PM" src="https://github.com/user-attachments/assets/bc08be10-9c46-48ce-9095-342d651d9eee" />